### PR TITLE
Add node-local-dns-cache to user-cluster-controller-manager

### LINKED ISSUE #3372: Move DNS handling into user cluster controller
**User Story**
All DNS specific resources should not be deployed using addons.
Addons should be optional - customer specific - resources.

DNS is a core component of a Kubernetes cluster.
With https://github.com/kubermatic/kubermatic/pull/3369 we now split logic across an addon & the controllers - Giving more potential for errors.

To avoid having logic split between addons & controllers + to ease [CIDR changes](https://docs.kubermatic.io/operation/control-plane/increase-service-pod-cidr/)
all DNS specific resources should be deployed by the UCCM (Potentially a own controller)

**Acceptance criteria**
- [ ] The DNS & nodelocal-dns-cache addons got removed
- [ ] CoreDNS & nodelocal-dns-cache are now being deployed by a own controller

### PR DESCRIPTION
**Pull request by moelsayed**
**What this PR does / why we need it**:
Adds node-local-dns-cache to user-cluster-controller-manager
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3372

**Special notes for your reviewer**:
The labels set are different from the ones on the coredns resources. But, they are the same as the ones currently deployed by the addon template. Once this is merged I will update them both to standard labels. 
**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
local-node-dns-cache is no longer a manifest addon. 
```

### CODE REVIEW

File: api/pkg/controller/user-cluster-controller-manager/resources/resources/node-local-dns/configmap.go
Comment (xrstf) [inline api/pkg/controller/user-cluster-controller-manager/resources/resources/node-local-dns/configmap.go]: reconcil**e**ModeValue

File: api/pkg/controller/user-cluster-controller-manager/resources/resources/node-local-dns/configmap.go
Comment (xrstf) [inline api/pkg/controller/user-cluster-controller-manager/resources/resources/node-local-dns/configmap.go]: What do we need this label for? The resource is not managed by the addon controller anymore, is it?

**Review by xrstf — CHANGES_REQUESTED** [review]
Feedback

File: api/pkg/controller/user-cluster-controller-manager/resources/resources/node-local-dns/configmap.go
Comment (moelsayed) [reply api/pkg/controller/user-cluster-controller-manager/resources/resources/node-local-dns/configmap.go]: it's related to k8s addon manager, not k8c's. It's in the original manifests, so I kept it around. Once this is merged, I will make a separate PR to clean and unify labels for all addons.

**Comment by moadqassem** [comment]
@moelsayed Can you please resolve the conflicts :)

**Comment by moadqassem** [comment]
/approve
/lgtm
/hold 

The tests are passing now, @xrstf if you agree on the addressed changes please remove the hold command.

**Review by xrstf — APPROVED** [review]
/hold cancel

**Pull request merged**
